components/brave_shields/browser/brave_shields_web_contents_observer.cc

```diff
@@ -94,7 +94,9 @@ namespace brave_shields {
 
 base::Lock BraveShieldsWebContentsObserver::frame_data_map_lock_;
 std::map<BraveShieldsWebContentsObserver::RenderFrameIdKey, GURL>
-    BraveShieldsWebContentsObserver::render_frame_key_to_tab_url;
+    BraveShieldsWebContentsObserver::frame_key_to_tab_url_;
+std::map<int, GURL>
+    BraveShieldsWebContentsObserver::frame_tree_node_id_to_tab_url_;
```

Comment:
Additional map for tracking frame tree node ids. They are available on the IO thread for frames/subframes instead of the pair procces_id/frame_id